firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteSerializer.java

```diff
@@ -648,7 +482,7 @@ public MutationResult decodeMutationResult(
     if (transformResultsCount > 0) {
       transformResults = new ArrayList<>(transformResultsCount);
       for (int i = 0; i < transformResultsCount; i++) {
-        transformResults.add(decodeValue(proto.getTransformResults(i)));
+        transformResults.add(FieldValue.valueOf(proto.getTransformResults(i)));
```

Comment:
Similarly: can we make transform results a model.ArrayValue so we don't have to convert?